src/org/gk/slicing/RevisionDetector.java

```diff
@@ -0,0 +1,442 @@
+	 *
+	 * @param oldInstance
+	 * @param newInstance
+	 * @param attrName
+	 * @return boolean (true if there is an addition or deletion in the attribute list, false if the lists are equal).
+	 * @throws InvalidAttributeException
+	 * @throws SQLException
+	 * @throws Exception
+	 */
+	private boolean additionOrDeletionInList(GKInstance oldInstance, GKInstance newInstance, String attrName)
+			throws InvalidAttributeException, SQLException, Exception {
+		Collection<GKInstance> newInstanceList = newInstance.getAttributeValuesList(attrName);
+		Collection<GKInstance> oldInstanceList = oldInstance.getAttributeValuesList(attrName);
+
+		// Constant time check to detect additions or deletions.
+		// Misses the case where the same number of unique additions and deletions have occurred.
+		// That case is detected by the iterator below.
+		if (newInstanceList.size() > oldInstanceList.size()
+		 || newInstanceList.size() < oldInstanceList.size())
+			return true;
+
+		// O(n^2) time check to detect additions or deletions.
+		for (Object instance : oldInstanceList) {
+			// If newInstanceList does not contain an instance in oldInstanceList,
+			// then either newInstanceList has a deletion, or oldInstanceList has an addition.
+			if (!newInstanceList.contains(instance)) {
+				for (Object inst : newInstanceList) {
+					Long oldInstanceDBID = ((GKInstance) inst).getDBID();
+					Long newInstanceDBID = ((GKInstance) inst).getDBID();
+
+					// An attribute has been added or deleted from the instance.
+					if (!oldInstanceDBID.equals(newInstanceDBID))
+						return true;
+				}
+			}
+		}
+
+		// The lists have the same elements.
+		return false;
+	}
+
+	/**
+	 * Check if an instance's summation attribute is revised.
+	 *
+	 * @param instance
+	 * @return boolean (true if summation is revised, false otherwise).
+	 * @throws InvalidAttributeException
+	 * @throws Exception
+	 *
+	 * @see {@link org.gk.model.Summation}
+	 */
+	private boolean isSummationRevised(GKInstance instance, MySQLAdaptor targetDBA)
+			throws InvalidAttributeException, Exception {
+		Collection<GKInstance> summations = instance.getAttributeValuesList(ReactomeJavaConstants.summation);
+		for (GKInstance summation : summations) {
+			// if a change in text is detected, then summation is considered revised.
+			if (attributesRevised(summation, getInstance(targetDBA, summation), ReactomeJavaConstants.text))
+				return true;
+		}
+
+		return false;
+	}
+
+	/**
+	 * Given a particular GKInstance, return the associated GKInstance from the "compare" slice.
+	 *
+	 * @param instance
+	 * @return GKInstance
+	 * @throws SQLException
+	 * @throws Exception
+	 */
+	private GKInstance getInstance(MySQLAdaptor dba, GKInstance instance) throws SQLException, Exception {
+		return getInstance(dba, instance.getDBID());
+	}
+
+	private GKInstance getInstance(MySQLAdaptor dba, Long DBID) throws SQLException, Exception {
+		return dba.fetchInstance(DBID);
```

Comment:
What is the advantage to have this method with only single line?

Reply:
Good point, it was a helper function, but with only one line doesn't do much *helping* per se. Fixed in [da57b5e](https://github.com/reactome/CuratorTool/pull/20/commits/da57b5eef5fec55923b43f6c5666583809a87fa5).